src/Farmer/Arm/ResourceGroup.fs

```diff
@@ -11,14 +11,15 @@ type DeploymentMode = Incremental|Complete
 
 /// Represents all configuration information to generate an ARM template.
 type ResourceGroupDeployment =
-    { Name: ResourceName
+    { Name: ArmExpression
```

Comment:
This is primarily because setting the domain requires a nested deployment and that nested deployment needs to know the name of the outer resource group. We considered changing the BuildResource function to take a context object including the resource group name but this seemed like the smaller change